src/sagas/combat-sagas.tsx

```diff
@@ -0,0 +1,193 @@
+import { takeEvery, put, all, select } from "redux-saga/effects";
+import {
+  ENTER_COMBAT,
+  CombatAction,
+  SELECT_PLAYER_COMBAT_ACTION,
+  SELECT_ENEMY_COMBAT_ACTION,
+  SelectEnemyCombatActionAction,
+  CombatActionType,
+  AlterEnemyHealthAction,
+  ALTER_ENEMY_HEALTH,
+  AlterMomentumAction,
+  ALTER_MOMENTUM
+} from "../player-actions/violence/types";
+import { AddMessageAction, ADD_MESSAGE } from "../message-feed/types";
+import { StoreState } from "../redux-common/store";
+import { enemyCombatantReducerDefaultState } from "../redux-common/default-store-state";
+import _ from "lodash";
+import { AlterHealthAction, ALTER_HEALTH } from "../resources/types";
+
+function* combatActionListener() {
+  yield takeEvery(SELECT_PLAYER_COMBAT_ACTION, resolveCombatTurn);
+}
+
+function* resolveCombatTurn() {
+  yield reportSelectedAction();
+  yield reportEnemyAction();
+  yield calculateDamage();
+  yield selectEnemyAction();
+}
+
+function* calculateDamage() {
+  const playerSelectedCombatAction = yield select((state: StoreState) =>
+    state ? state.playerSelectedCombatAction : null
+  );
+  const enemySelectedCombatAction = yield select((state: StoreState) =>
+    state ? state.enemySelectedCombatAction : null
+  );
+  const momentum = yield select((state: StoreState) =>
+    state ? state.momentum : 0
+  );
+
+  const playerAction: CombatAction = playerSelectedCombatAction;
+  const enemyAction: CombatAction = enemySelectedCombatAction;
+
+  const counter = calculateCounter(playerAction, enemyAction);
+
+  if (counter === -2) {
+    yield sendMessage(`${playerAction.name} counters ${enemyAction.name}`);
+  } else if (counter === 2) {
+    yield sendMessage(`${enemyAction.name} counters ${playerAction.name}`);
+  }
+
+  const outcomeValue =
+    enemyAction.speedValue - playerAction.speedValue + counter + momentum;
+
+  if (outcomeValue < 0 && playerAction.type === CombatActionType.Offense) {
+    yield dealDamage(playerAction.damage, "Enemy");
+    yield alterMomentum(-1);
+    yield sendMessage(
+      `Your attack landed dealing ${playerAction.damage} damage. You gain 1 momentum`
+    );
+  } else if (
+    outcomeValue > 0 &&
+    enemyAction.type === CombatActionType.Offense
+  ) {
+    yield dealDamage(enemyAction.damage, "Player");
+    yield alterMomentum(1);
+    yield sendMessage(
+      `Enemy attack landed dealing ${enemyAction.damage} damage. You lose 1 momentum`
+    );
```

Comment:
These 3 can probably be abstracted into common function just taking some variables, not sure it's worth it though

Reply:
aha ye i did exactly that then reverted cos i ended up deciding 95% of the functions were 'helpers' and couldnt really differentiate which should go in which file. hate how bloated this file is tho